### PR TITLE
SONARJAVA-2333 Update rule descriptions

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S2164_java.html [hunk 1]
```diff
@@ -16,6 +16,12 @@
 
 double d = (double)a + (double)b;
 </pre>
+<h2>Exceptions</h2>
```
Comment (m-g-sonar) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S2164_java.html]: This is not yet implemented, and part of [SONARJAVA-2323](https://jira.sonarsource.com/browse/SONARJAVA-2323).

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S2757_java.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
-<p>The use of operators pairs ( <code>=+</code>, <code>=-</code> or <code>=!</code> ) where the reversed, single operator was meant (<code>+=</code>,
-<code>-=</code> or <code>!=</code>) will compile and run, but not produce the expected results.</p>
+<p>The use of operators pairs ( <code>=+</code>, <code>=-</code> or <code>=\!</code> ) where the reversed, single operator was meant (<code>+=</code>,
+<code>-=</code> or <code>\!=</code>) will compile and run, but not produce the expected results.</p>
```
Comment (m-g-sonar) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S2757_java.html]: The escape of `!` is required on JIRA, but should not be part of the code here.

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S2164_java.html (hunk 1 above)
Comment (benzonico) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S2164_java.html]: taking care of implementing it.

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S2757_java.html (hunk 2 above)
Comment (erichirlemann) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S2757_java.html]: Issue recorded as https://jira.sonarsource.com/browse/RULEAPI-464

Comment (benzonico) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S2757_java.html]: Bug reported back to rule-api : https://jira.sonarsource.com/browse/RULEAPI-464

**Pull request merged**